lib/rihanna/job.ex

```diff
@@ -177,6 +177,65 @@ defmodule Rihanna.Job do
   end
 
   @doc false
+  def delete_by(opts) do
+    ids_to_delete =
```

Comment:
This is a tricky one since we _must_ load the terms into erlang to do the filtering. However, this is going to take a lot of memory and could be an expensive operation if there are 100,000s of jobs in the table.

Can we do it in batches?

Reply:
Yes, we need to do this in batches otherwise this function could bring a node down by consuming all the memory.

Reply:
This does make me wonder if we should store the module name using a string, it makes this sort of thing much easier and more efficient.

It would also resolve the problem I have in production where I can't write health check queries or manually look using SQL because all the useful job data is opaque.

Reply:
Module name and function name could both be strings. This would be a breaking change but we could certainly do it.

Argument must continue to be an opaque term.

Reply:
Makes sense. Could be the same release as the new column.

Reply:
Agreed. I'd also like to get a `queue` column in there while we're at it.

Reply:
Great idea

Reply:
@samphilipd what is a opaque term?